src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Serialization.cs

```diff
@@ -181,26 +181,49 @@ public void WriteTo(ObjectWriter writer)
                     }
                 }
             }
+
+            // sortedNodes is an array of Node instances which is often sorted by Node.Name by the caller. This method
+            // produces a sequence of spans within sortedNodes for Node instances that all have the same Name, allowing
+            // serialization to record the string once followed by the remaining properties for the nodes in the group.
+            static IEnumerable<ReadOnlyMemory<Node>> GroupByName(ImmutableArray<Node> sortedNodes)
+            {
+                if (sortedNodes.IsEmpty)
+                    yield break;
+
+                var startIndex = 0;
+                var currentName = sortedNodes[0].Name;
+                for (var i = 0; i < sortedNodes.Length; i++)
```

Comment:
consider starting at 1.  i found it slightly harder to convince myself of correctness in the case where we started on hte same item we'd already looked at and cached information for.

Reply:
➡️ Done